src/System.Runtime/tests/System/String.cs

```diff
@@ -702,6 +703,13 @@ public static void TestJoin()
     {
         String s;
 
+        // String Array
+        s = String.Join("$$", new String[] { null }, 0, 1);
+        Assert.True(s == "");
+
+        s = String.Join("$$", new String[] { null, "Bar", null }, 0, 3);
+        Assert.True(s == "$$Bar$$");
```

Comment:
Instead of `Assert.True` you can use `Assert.Equal` here and below. It prints a much better message when the test case fails.